### ISSUE TITLE
Listing repos for an org that has multiple page seem to go into a loop forever

### ISSUE THREAD

**Issue by nicolasgarnier** [DirectMyFile/github.dart#53]
On my app [GitHub Issue Mover](https://github-issue-mover.appspot.com/) it seems that when trying to list the repos for an org that has lots of repos (multiple pages), like for example the `dart-lang` org or `GoogleChrome` or `GoogleCloudSamples` or `google` for instance The page 2 is being pulled forever when calling this method:

    gitHub.repositories.listUserRepositories(orgName).toList()

When I look at the Network in Chrome dev tab I see the following page downloaded many may times:

Request URL:https://api.github.com/user/1778935/repos?type=owner&sort=full_name&direction=asc&page=2?type=owner&sort=full_name&direction=asc
Request Method:GET

The reponse does contains the following header:
```
Link:<https://api.github.com/user/1342004/repos?type=owner&sort=full_name&direction=asc&page=2>; rel="next", <https://api.github.com/user/1342004/repos?type=owner&sort=full_name&direction=asc&page=13>; rel="last"
```

But Page 2 is re-downloaded always.

I did not change anything in the app for month but surely App Engine deployed a new version so it might have downloaded a new version of `github.dart`. Any idea what could be causing this?

Cheers!

**Comment by kaendfinger** [comment]
Unable to reproduce on the VM. I'm investigating this however.

**Comment by kaendfinger** [comment]
@nicolasgarnier I tried to reproduce but I am unable to.

Try this page: http://github.directcode.org/demos/repos.html?user=dart-lang

I pushed an update just in case something weird is going on, let me know if it works after that.

**Comment by nicolasgarnier** [comment]
actually the issue couldn't have been a coming from a new version of GitHub (I take back what I said) because I deployed the app with a pubspec.lock so the same version of github.dart would have been pulled every time. It's likely a change in the gitHub API behavior... not sure what though.

I tried deploying my app locally with updated version of all dependencies and things work fine locally. I'm deploying on prod now to see if this fixes things :). Anyways not an issue with github.dart so I'm closing this.

Thanks!
